worldedit-core/src/main/java/com/fastasyncworldedit/core/Fawe.java

```diff
@@ -140,14 +147,27 @@ private Fawe(final IFawe implementation) {
         }, 0);
 
         TaskManager.taskManager().repeat(timer, 1);
-
-        clipboardExecutor = new KeyQueuedExecutorService<>(new ThreadPoolExecutor(
+        uuidKeyQueuedExecutorService = new KeyQueuedExecutorService<>(new ThreadPoolExecutor(
                 1,
                 Settings.settings().QUEUE.PARALLEL_THREADS,
                 0L,
                 TimeUnit.MILLISECONDS,
                 new LinkedBlockingQueue<>(),
-                new ThreadFactoryBuilder().setNameFormat("FAWE Clipboard - %d").build()
+                new ThreadFactoryBuilder().setNameFormat("FAWE UUID-key-queued - %d").setThreadFactory(new ThreadFactory() {
+                    private final ThreadFactory defaultFactory = Executors.defaultThreadFactory();
+
+                    @Override
+                    public Thread newThread(@Nonnull Runnable r) {
+                        return defaultFactory.newThread(() -> {
+                            IS_UUID_KEYED_EXECUTOR_THREAD.set(true);
```

Comment:
What's the idea here? I'd assume every thread in this thread pool is always a uuid keyed executor thread? Would it be enough to subtype `Thread` and do an instanceof check where needed?

Reply:
It needs to be accessible from runners within the thread itself. I didn't want to offload to an async thread from main thread needlessly to then run on this thread again. I suppose checking if the current thread is an instance of subtype would be cleaner